src/server/game/Scripting/ScriptMgr.cpp

```diff
@@ -913,17 +920,6 @@ class SpecializedScriptRegistry<ScriptType, true>
             sScriptRegistryCompositum->SetScriptNameInContext(script->GetName(),
                 sScriptMgr->GetCurrentScriptContext());
         }
-        else
```

Comment:
So you are now leaking memory when script isnt assigned to anything?

Reply:
Well thats exactly what i do, if you as example create a new script via hotswap and you don't got the database entry yet, the script will be removed. Thats why i dont delete them. Better idea how i can get them back if they're removed? 🤔

Reply:
But you don't need to. You seem to misunderstand how hotswap works, it completely reloads all scripts dlls (that means replacing all the code, not just AIs) and when new one is loaded, all AddSC functions are called

Reply:
yeah but thats the problem, if you as example first save the new script, which triggers the hotswap system and calls all AddSC functions, afterwards you put your scriptname assignment into the database and you want to reload, but there isn't any script which is assigned to this scriptname

Reply:
So it this always a issue?

Reply:
> if you as example first save the new script, which triggers the hotswap system and calls all AddSC functions, afterwards you put your scriptname assignment into the database and you want to reload, but there isn't any script which is assigned to this scriptname

what about triggering again HotSwap script unloading and re-loading then in that case ?

Reply:
Could do that, but in that case every reloading method has to call that method. Thats an option too.

Reply:
No, it's unaccetpable to remove the deferred deletion of the script.
Why you can't keep it?`It is still valid to delete the script in case no database scriptname was found.